src/Diagnostics.DataProviders/DataProviderLogDecorator.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Diagnostics.Logger;
+using Diagnostics.ModelsAndUtils.Models;
+
+namespace Diagnostics.DataProviders
+{
+    internal class DataProviderLogDecorator : IKustoDataProvider, IGeoMasterDataProvider, ISupportObserverDataProvider
```

Comment:
Do you forsee this class to explode as it has methods from all the data providers?

Reply:
Since kusto and observer are following that parameterized approach (Kusto.ExecuteQuery and Observer.GetResource), I don't see this class blowing up in the near future. Hopefully geomaster will follow the same approach.